AU2/plugins/custom_plugins/WantedPlugin.py

```diff
@@ -172,50 +169,74 @@ def on_page_generate(self, htmlResponse) -> List[HTMLComponent]:
             )
         if wanted_police:
             rows = []
-            for player in wanted_police:
-                (_, crime, redemption) = wanted_manger.get_wanted_player_crime_info(player, now)[1]
+            if police_ranks_enabled:
+                default_rank = GENERIC_STATE_DATABASE.arb_state.get(
+                    "PolicePlugin", {}).get("PolicePlugin_default_rank", DEFAULT_POLICE_RANK)
+                ranks = GENERIC_STATE_DATABASE.arb_state.get("PolicePlugin", {}).get("PolicePlugin_ranks", DEFAULT_RANKS)
+            for player_id in wanted_police:
+                player = ASSASSINS_DATABASE.get(player_id)
+                rank = "Police"
+                if police_ranks_enabled:
+                    rank_no = police_rank_manager.get_relative_rank(player) + default_rank
+                    try:
+                        rank = ranks[rank_no]
+                    except IndexError:
+                        # Lets the user know if the ranks break due to PolicePlugin.generate_pages not generating new ranks in time.
+                        # Could maybe be avoided with a refactor of how ranks work, but I really don't want to do that now

```

Comment:
If you don't want to do this now, please do the following:
1) Open an issue on GitHub explaining the issue
2) Leave a TODO with a link to the issue 

This can be solved by moving the logic into the police rank manager, so make note of that

Reply:
I did try to move it into police rank manager once, but it required so much policeplugin database access that I didn't want to change everything. Is it reasonable to keep the db structure the same whil;e having all the code in policerankmanager?